node/network/bridge/src/multiplexer.rs

```diff
@@ -68,19 +69,31 @@ impl RequestMultiplexer {
 			})
 			.unzip();
 
+		// Ok this code is ugly as hell, it is also a hack, see #2842. But it works and is executed
+		// on startup so, if anything is wrong here it will be noticed immediately.
 		let index = receivers.iter().enumerate().find_map(|(i, (p, _))|
 			if let Protocol::StatementFetching = p {
 				Some(i)
 			} else {
 				None
 			}
-		).expect("Statement fetching must be registered. qed.");
+		).expect("Statement fetching must be registered.");
 		let statement_fetching = Some(receivers.remove(index).1);
 
+		let index = receivers.iter().enumerate().find_map(|(i, (p, _))|
+			if let Protocol::DisputeSending = p {
+				Some(i)
+			} else {
+				None
+			}
+		).expect("Dispute sending must be registered.");
```

Comment:
A few too many expects imho that will be transformed into footguns in the future™. I'd prefer early returns and a warning/error message trace.

Reply:
I did not bother here, because the code is
a) to be abandoned soon™
b) more importantly - executed at startup and not in some seldomly hit code path.

Anyhow early returns might be fine as well, will check.